### PR TITLE
fetching indicator data using the new endpoint

### PR DESCRIPTION
**Pull request by emre2038**
## Description
* removing children indicator data from `all_details` API endpoint
* fetching the children indicator data using the new endpoint `children-indicators`
* removing webflow interactions from data mapper and managing expanding/collapsing the menu elements by javascript
* mapchip & data mapper goes into loading state when navigating between geographies

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-191

## How is it tested automatically?
added test steps into `__tests__/gui/data_mapper/data_mapper.js`

## How should a reviewer test it locally
* confirm that the data mapper works the same way it used to worked
* confirm that choropleth & filtering works correctly
* confirm that rich data charts & filtering works correctly
* confirm that multi-version structure is not broken by testing against SANEF profile
* confirm that choropleth is not broken when navigating between geographies

## Screenshots
image

### Updated
* `src/js/controller.js` & `src/js/setup/choropleth.js` to use the `data` property of `children-indicators` instead of the `child_data` property of `all_details` when displaying choropleth
* `src/js/elements/mapchip/legend.js` & `src/js/elements/mapchip/mapchip.js` for the mapchip to go into loading state when navigating between geographies
* `src/js/elements/menu.js` to adapt webflow changes to remove webflow interactions from data mapper elements

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

**Comment by goyal1092** [comment]
@emre2038  Was able to find couple of issues in Datamapper: https://purple-surfboard-fd1.notion.site/Bug-List-39e4a7773be742628093d99980475470

* rich data charts & filtering works correctly and it worked perfectly
* choropleth & filtering works correctly
* was not able to test :  multi-version structure is not broken by testing against SANEF profile

**Comment by emre2038** [comment]
> * was not able to test :  multi-version structure is not broken by testing against SANEF profile
@goyal1092 you need to set the API url so it works against staging

**Review by michaelglenister — COMMENTED** [review]
Looks good so far. I added some minor points. And then just fixing CI

File: __tests__/gui/common_cy_functions/general.js [hunk 1]
```diff
@@ -43,11 +44,19 @@ export function setupInterceptions(profiles, all_details, profile, themes, point
 
     cy.intercept('/api/v1/profile/8/geography/ZA/themes_count/?version=test&format=json', (request) => {
         request.reply({
-            statusCode: 200,
+            statusCode: 201,
```
[the remark this answers is not in the document]
Comment (emre2038) [reply __tests__/gui/common_cy_functions/general.js]: updated all the requests so they return 200

File: src/js/elements/menu.js [hunk 2]
```diff
@@ -86,50 +86,77 @@ export function loadMenu(dataMapperMenu, data, subindicatorCallback) {
...
                 const childWrapper = $(newIndicator).find('.data-category__h3_wrapper');
 
                 let subindicators = detail.metadata.groups.filter((group) => group.name === detail.metadata.primary_group)[0];
+                maxHeight += triggerHeight;
                 addSubIndicators(childWrapper, category, subcategory, indicator, subindicators, indicators, detail);
+                //$(newIndicator).find('.data-category__h3_content--v2').css('max-height', 1000);
             }
         }
+
+        //$(newSubCategory).find('.data-category__h2_content--v2').css('max-height', maxHeight);
```
[the remark this answers is not in the document]
Comment (emre2038) [reply src/js/elements/menu.js]: thanks and nice catch. removed

File: __tests__/gui/multi_boundary_versions/children_indicators_2011.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                  "dataset":1034,
+                  "name":"categories",
+                  "can_aggregate":true,
+                  "can_filter":true
+                }
+              ]
+            },
+            "content_type":"indicator",
+            "dataset_content_type":"quantitative",
+            "version":"2016 with wards",
```
[the remark this answers is not in the document]
Comment (emre2038) [reply __tests__/gui/multi_boundary_versions/children_indicators_2011.json:40]: I just copied it from sanef profile I think a while ago when we implemented multiple version. do you think it creates confusion?

[the remark this answers is not in the document]
Comment (michaelglenister) [reply __tests__/gui/multi_boundary_versions/children_indicators_2011.json:40]: I think this is okay for now, we can update this test data in future if needed

**Review by michaelglenister — COMMENTED** [review]
The code looks good and the test coverage is good. If you can just fix the CI. Thanks :smile:

**Pull request merged**